C++17 detection fails with conda's clang

this patch can be removed once clang 7 is in defaults
diff --git a/configure.ac b/configure.ac
index 1353248..936d2b8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -32,9 +32,6 @@ AX_VALGRIND_CHECK()
 dnl Check for required libraries
 AC_LANG([C++])
 
-dnl We use some C++17 features such as std::optional
-AX_CXX_COMPILE_STDCXX_17()
-
 ACX_PTHREAD([], [AC_MSG_ERROR([pthread not supported])])
 
 AC_CHECK_HEADERS([arb.h], , AC_MSG_ERROR([Arb headers not found]))

